mock/mock.go

```diff
@@ -527,6 +527,47 @@ func (m *Mock) AssertNotCalled(t TestingT, methodName string, arguments ...inter
 	return true
 }
 
+// IsMethodCallable checking that the method can be called
+// If the method was called more than `Repeatability` return false
+func (m *Mock) IsMethodCallable(t TestingT, methodName string, arguments ...interface{}) bool {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	var isCallable = false
+	for _, v := range m.ExpectedCalls {
+		if v.Method != methodName {
+			continue
+		}
+		if len(arguments) != len(v.Arguments) {
+			continue
+		}
+		if v.Repeatability < v.totalCalls {
+			continue
+		}
+		if isArgsEqual(v.Arguments, arguments) {
+			isCallable = true
+			break
```

Comment:
Creating a variable (`isCallable` in this case) and then overwriting it in a narrower scope is _usually_ a sign that you should be returning instead.   Sometimes it's a sign that you need to extract the logic out to a separate function and set the variable to the return of that function.

In this case, it's perfectly reasonable to just `return true` here and delete the `isCallable` variable.

Reply:
Agreed, we can just `return true` at line 551 and have no need for the `isCallable` variable.